### ISSUE TITLE
Datadog client does not support querying scalar values using `api/v2/query/scalar` 

### ISSUE THREAD

**Issue by vadasambar** [DataDog/datadog-api-client-go#1233]
**Note:**
If you have a feature request, you should [contact support](https://docs.datadoghq.com/help/) so the request can be properly tracked.

**Noted.** 

**Is your feature request related to a problem? Please describe.**
I want to be able to get scalar value for a query so that I can use it to evaluate SLO/SLI ([related](keptn/integrations#23)). 

Based on my understanding:

![image](https://user-images.githubusercontent.com/34534103/144263392-259fd017-cd53-47e4-820a-dc29a8caacd6.png)

URL:  `POST | https://app.datadoghq.com/api/v2/query/scalar`

#### Request
<details>
<summary> </summary>

```json
{
   "meta":{
      "dd_extra_usage_params":{
         
      }
   },
   "data":[
      {
         "type":"scalar_request",
         "attributes":{
            "formulas":[
               {
                  "formula":"query1"
               }
            ],
            "queries":[
               {
                  "query":"avg:system.load.1{*}.rollup(avg, 1200)",
                  "data_source":"metrics",
                  "name":"query1",
                  "aggregator":"avg"
               }
            ],
            "from":1638368994000,
            "to":1638372594000
         }
      }
   ],
   "_authentication_token":"XXX"
}
```
</details>

#### Response
<details>
<summary></summary>

```json
{
   "meta":{
      "res_type":"scalar",
      "responses":[
         
      ]
   },
   "data":[
      {
         "type":"scalar_response",
         "attributes":{
            "columns":[
               {
                  "type":"number",
                  "meta":{
                     "unit":null
                  },
                  "values":[
                     0.8087170141667361
                  ],
                  "name":"query1"
               }
            ]
         }
      }
   ]
}
```
</details>

I want to obtain this value programmatically using the go client library. The v1 version of the client [only supports querying `Time Series` values](https://github.com/DataDog/datadog-api-client-go/blob/v1.6.0/api/v1/datadog/docs/MetricsApi.md#QueryMetrics). 
If I use the same Datadog Notebook as above, 
URL: `POST | https://app.datadoghq.com/api/v2/query/timeseries`
#### Request
<details>
<summary></summary>

```json
{
   "meta":{
      "dd_extra_usage_params":{
         
      }
   },
   "data":[
      {
         "type":"timeseries_request",
         "attributes":{
            "queries":[
               {
                  "query":"avg:system.load.1{*}.rollup(avg, 1200)",
                  "data_source":"metrics",
                  "name":"query1"
               }
            ],
            "from":1638368994000,
            "to":1638372594000,
            "interval":20000,
            "formulas":[
               {
                  "formula":"query1"
               }
            ]
         }
      }
   ],
   "_authentication_token":"XXX"
}
```
</details>

#### Response
<details>
<summary></summary>

```json
{
   "meta":{
      "res_type":null,
      "responses":[
         {
            "is_millisecond":true,
            "res_type":"time_series",
            "from_date":1638368994000,
            "to_date":1638372594000,
            "interval":1200000
         }
      ]
   },
   "data":[
      {
         "type":"timeseries_response",
         "attributes":{
            "series":[
               {
                  "unit":null,
                  "group_tags":[
                     
                  ],
                  "query_index":0
               }
            ],
            "times":[
               1638370800000,
               1638372000000
            ],
            "values":[
               [
                  0.811864,
                  0.80557
               ]
            ]
         }
      }
   ]
}
```
</details>

If you take an average of the values in the `Time Series` query response above (0.80557 + 0.811864) / 2 = 0.808717 which is the value you will get in the `Query Value` aka scalar query response. 

This means, if I had to use the `Time Series` query API to mimic the behavior of `Query Value`, I will have to do the above calculation. 

**Describe the solution you'd like**
I want Datadog client to support `api/v2/query/scalar` so that I can get the same behavior as the UI.

**Describe alternatives you've considered**
No real alternatives as of now. If I had to mimic the behavior for `Query Value` using the time series API, I would have to implement the calculation part by myself which Datadog does for me if I am using the UI.

**Additional context**
keptn/integrations#23
Please correct me if I misunderstood something :pray:

**Comment by vadasambar** [comment]
I have raised a ticket with Datadog support as well. Will update here if I get a response.

**Comment by nikita8** [comment]
@vadasambar Were you able to get any response on this? I was looking for something similar in ruby.

**Comment by vadasambar** [comment]
@nikita8, 
I need to re-confirm this but it seems to be getting the last value in the timeseries. If you can explain your use case more, maybe I can help you.

**Comment by nikita8** [comment]
> @nikita8, I need to re-confirm this but it seems to be getting the last value in the timeseries. If you can explain your use case more, maybe I can help you.

I was looking for an API to get the overall p95 and p50 value for a given range . For eg: p95 runtime for last 3 months. With the current metric API, I can apply the rollup only for max 1 month to the timeseries. So there is no way for me to reduce the values for time range greater than 1 month while making the API request.

**Comment by therve** [comment]
Fixed in #1798
